mobile/lib/services/sync.service.dart

```diff
@@ -566,62 +559,62 @@ class SyncService {
         .findAll();
     assert(inDb.isSorted(Asset.compareByChecksum), "inDb not sorted!");
     final int assetCountOnDevice = await ape.assetCountAsync;
-    final List<Asset> onDevice =
-        await _hashService.getHashedAssets(ape, excludedAssets: excludedAssets);
-    _removeDuplicates(onDevice);
-    // _removeDuplicates sorts `onDevice` by checksum
-    final (toAdd, toUpdate, toDelete) = _diffAssets(onDevice, inDb);
-    if (toAdd.isEmpty &&
-        toUpdate.isEmpty &&
-        toDelete.isEmpty &&
-        album.name == ape.name &&
-        ape.lastModified != null &&
-        album.modifiedAt.isAtSameMomentAs(ape.lastModified!)) {
-      // changes only affeted excluded albums
+    await for (final onDevice in _hashService.getHashedAssets(ape)) {
+      _removeDuplicates(onDevice);
+      // _removeDuplicates sorts `onDevice` by checksum
+      final (toAdd, toUpdate, toDelete) = _diffAssets(onDevice, inDb);
```

Comment:
This is the issue:
You compute a diff between the entire DB and a small batch from assets on device. The diff mechanism can't work that way. You'd either need to accumulate all device assets first (loosing the stream functionality for syncing...) OR you'd need to select the appropriate portion from the DB to compare to (which is challenging to get right, but possibly!)